src/Forms/NumericField.php

```diff
@@ -149,28 +178,36 @@ public function Value()
         return $formatter->format($this->value, $this->getNumberType());
     }
 
-    public function setValue($value, $data = null)
+    public function getValueForValidation(): mixed
     {
-        $this->originalValue = $value;
-        $this->value = $this->cast($value);
-        return $this;
+        $value = $this->value;
+        if (is_numeric($value) && is_string($value)) {
+            return $this->cast($value);
+        }
+        return $value;
     }
 
     /**
      * Helper to cast non-localised strings to their native type
```

Comment:
This doesn't seem to be what's happening anymore - we're casting to string after casting away from string, apparently.

Reply:
Updated